### ISSUE TITLE
ray tracing vs. casting

### ISSUE THREAD

**Issue by banesullivan** [pyvista/pyvista#478]
The `ray_trace` filter on `PolyData` meshes is actually "ray casting". We should rename this method and also add a method for "ray tracing" per https://blog.kitware.com/ray-casting-ray-tracing-with-vtk/

> we only ‘cast’ a single ray, test for its intersection with objects, and retrieve information regarding the intersection. Ray-tracing, on the other hand, is more physically accurate, as it applies laws of physics (e.g., reflection, refraction, attenuation, etc.) to the rays to ‘trace’ (i.e., follow) that ray and its derivative rays. However, ray-casting is the natural precursor to ray-tracing, as it tells us with what part of which object the ray intersects and provides all necessary information to cast subsequent rays.

Related to https://github.com/pyvista/pyvista-support/issues/80

**Comment by akaszynski** [comment]
I’m fine with this as long as you leave a warning and a redirect as I’m using this in my closed source project.

**Comment by banesullivan** [comment]
Yeah... I figured there would be some backward compatibility issues as `ray_trace` has been in there for a very long time. 

I'll see what I can do to keep backward compatibility. Perhaps just default the new ray-tracing method to be the first cast ray so the two methods have the same result with default parameters... we'll see when I (or whoever) gets around to implementing this. 

I have no need for this at the moment but came across the discrepancy and figured we should outline this for the future.

**Comment by banesullivan** [comment]
We settled on ray_tracing long ago
